samples/client/petstore/php/OpenAPIClient-php/tests/ObjectSerializerTest.php

```diff
@@ -155,13 +155,11 @@ public function provideQueryParams(): array
 
         return [
             // style form
-            // color=
             'form empty, explode on' => [
-                '', 'color', 'string', 'form', true, 'color=',
```

Comment:
This change really bothers me. OpenAPI spec clearly states that empty query value with `form` serialization must be encoded as `color=`.  Check [that table of style examples](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#style-examples). However after some investigation I noticed Swagger Live Editor doesn't send query param when it's empty. Maybe this behaviour is because of some front UI validation, we cannot check what query could be when clients code do pass empty values.

Does it really breaks something for you, when you receive `?color=&foobar=notempty` kinda values??? Your server implementation should be prepared for requests beside client PHP library(Postman, Insomnia, pure Curl etc). You cannot force user to use client SDK if he doesn't want to, then he can send weird values anyway.

Reply:
Our API doesn't work with empty value it returns error 
e.g.
request 
```
https://api.*.com/crm/v3/objects/contacts?limit=10&after=&properties=website&properties=phone
```
Response
```
{
    "status": "error",
    "message": "Invalid paging token provided.(it means after)",
    "correlationId": "",
    "context": {
        "after": [
            "Optional[]"
        ]
    },
    "category": "VALIDATION_ERROR"
}
```
And before [these changes](https://github.com/OpenAPITools/openapi-generator/pull/11225) 
The Generator doesn’t put empty value
e.g.
request was without after
```
https://api.*.com/crm/v3/objects/contacts?limit=10&properties=website,phone
```
I understand the OpenAPI spec says opposite but if you check previous version you see that 
it's critical for us 
if it's critical for you I can create an Additional option for it or something like this 
CC @ybelenko

Reply:
> it's critical for us

You can use modified mustache template then. It's one line of code which you already mentioned here. @wing328 what do you think? We cannot change default behaviour for each user to keep their integrations safe.

> Our API doesn't work with empty value it returns error e.g.

So, you're not an API provider, right?

Reply:
I am API provider but we cannot change it in our side 
we have really big API and we cannot change it
Why in 5.4 or lower versions it doesn't put empty value to query string ?
@ybelenko @wing328

Reply:
> Why in 5.4 or lower versions it doesn't put empty value to query string ?

Because query serialization feature has been pure implemented at the point of time. I would say that it's wasn't implemented at all. Our PHP client is far behind latest OpenAPI spec. Anyway, if somebody from our PHP technical committee or core members will say that I understand current spec wrong then I change unit tests and implementation. No hard feelings, I just tried to make it as close to spec as possible.

Regarding your API spec. Let's get this straight, if we can omit these query params, then they are optional/ not required, right? So SDK users can pass `null` or `''`(empty string) and your server should handle it well, ignore it without throwing errors. When you use client SDK how do your execute API method? Could you drop the actual client code. Maybe we can add some `if` condition to API template to exclude query params when they're not required.

If these params are required client SDK shouldn't even make HTTP request with empty values, but throws exception immediately.

Reply:
Right , these params are not required (optional)
I have to say that we use not only php generator 
We also use typescript, typescript-node, ruby and python generators and anyone of these doesn’t pass empty values
You said `Maybe we can add some if condition to API template to exclude query params when they're not required.`
It is suit for us and I can implement it here

for example code of typescript client 
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript/api/api.mustache#L65
@ybelenko

Reply:
Hi @ybelenko 
Could you check it?
https://github.com/OpenAPITools/openapi-generator/pull/12120